src/Microsoft.AspNetCore.Server.Kestrel/Internal/Infrastructure/KestrelThread.cs

```diff
@@ -320,6 +328,18 @@ private void OnPost()
             } while (wasWork && loopsRemaining > 0);
         }
 
+        private void OnHeartbeat(UvTimerHandle timer)
+        {
+            Walk(ptr =>
+            {
+                var now = _engine.Libuv.hrtime();
+                //var now = DateTime.UtcNow.Ticks;
```

Comment:
DateTimeHeaderValueManager thing updates its time object every second; could add an actual DateTime property to it (as well as byte and string) and use that?